src/App.tsx

```diff
@@ -14,11 +15,112 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const App: React.FC = () => {
+  const [isStarted, setIsStarted] = useState(false);
+  const [reverse, setReverse] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+  const [lengthValue, setLengthValue] = useState(0);
+
+  const goods = [...goodsFromServer].filter(good => good.length >= lengthValue);
+  const lengthSelector = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+  const reset = () => {
+    setReverse(false);
+    setSortBy('');
+    setLengthValue(0);
+  };
```

Comment:
you read about [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback)
Optional )